lib/spack/spack/package.py

```diff
@@ -1808,6 +1808,12 @@ def do_test(self, dirty=False):
         self.tested_file = self.test_suite.tested_file_for_spec(self.spec)
         fsys.touch(self.test_log_file)  # Otherwise log_parse complains
 
+        if self.spec.external and not test_externals:
+            with open(self.test_log_file, 'w') as ofd:
+                ofd.write('Testing package {0}\n'
+                          .format(self.test_suite.test_pkg_id(self.spec)))
+            return
```

Comment:
I'm not super familiar with this part of the code, so just asking. Here it seems to me that we are in a situation where:
1. The spec is external
2. We don't want to test external packages

Why do we write a text file that says we are testing the package?

Reply:
Hmm.  Would it be better if I output that the test is being skipped?

Reply:
@tldahlgren The PR is fine with me.